src/wallet/walletdb.cpp

```diff
@@ -310,6 +312,11 @@ ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue,
                 strErr = "Error reading wallet database: LoadKey failed";
                 return false;
             }
+            if (!pwallet->GetLegacyScriptPubKeyMan()->LoadKey(key, vchPubKey))
+            {
+                strErr = "Error reading wallet database: LegacyScriptPubKeyMan::LoadKey failed";
+                return false;
+            }
         } else if (strType == DBKeys::MASTER_KEY) {
```

Comment:
Maybe leave a comment in this block that the pubkeyman has no use for this key, it's just an encryption key. I was tripped up on this. Bad naming! bad!

Reply:
Done